atlassian/connector.go

```diff
@@ -0,0 +1,76 @@
+package atlassian
+
+import (
+	"fmt"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/interpreter"
+	"github.com/amp-labs/connectors/common/paramsbuilder"
+	"github.com/amp-labs/connectors/common/urlbuilder"
+	"github.com/amp-labs/connectors/providers"
+)
+
+type Connector struct {
+	Client  *common.JSONHTTPClient
+	BaseURL string
+	Module  string
+	// workspace is used to find cloud ID.
+	workspace string
```

Comment:
Workspace would be domain name.
CloudID is associated with Workspace and is used in Rest URLs.
The API that we call has list of these 2 values (ignoring extra).

In all honesty, I am a bit confused. You are right that we should better use workspace instead of cloudID conceptually.
If we already know cloudID before calling constructor using `WithMetadata` why not pass it via `WithWorkspace`.

But then `GetPostAuthInfo` is sort of like appendix. Maybe to switch things up. **CloudID** will be renamed to **workspace** and **workspace** to **domain**. Resulting in the code like below:
```go
atlassian.NewConnector(
  WithWorkspace(actuallyCloudID)
  WithMetadata(map[string]string{"domain": userDomain})
)
```
Then we can use domain name such that `GetPostAuthInfo` which will return 
```go
map[string]string{
  "workspace": cloudID,
}
```

What do you think?